src/contexts/settingsContext.tsx

```diff
@@ -29,20 +29,35 @@ export interface SettingsContextValue {
   setTimestampPrecisionMode: (status: OnOff) => void
 }
 
+const systemThemeQuery = matchMedia('(prefers-color-scheme: dark)')
+
+//could be `no-preference` so default is `light`'
+const currentSystemTheme = () => (systemThemeQuery.matches ? 'dark' : 'light')
+
+const initialSystemTheme: ThemeType = currentSystemTheme()
+
 export const SettingsContext = createContext<SettingsContextValue>({
-  theme: 'light',
+  theme: initialSystemTheme,
   switchTheme: () => null,
   timestampPrecisionMode: 'off',
   setTimestampPrecisionMode: () => null
 })
 
 export const SettingsProvider = ({ children }: { children: ReactNode }) => {
-  const [themeName, setThemeName] = useStateWithLocalStorage<ThemeType>('theme', 'light')
+  const [themeName, setThemeName] = useStateWithLocalStorage<ThemeType>('theme', initialSystemTheme)
   const [timestampPrecisionMode, setTimestampPrecisionMode] = useStateWithLocalStorage<OnOff>(
     'timestampPrecisionMode',
     'off'
   )
 
+  const themeHandler = () => setThemeName(currentSystemTheme())
+
+  useEffect(() => {
+    systemThemeQuery.addEventListener('change', themeHandler)
+    return () => systemThemeQuery.removeEventListener('change', themeHandler)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Wrapping the themeHandler in a `useCallback` avoids to disable the linter for this line :)

```ts
  const themeHandler = useCallback(() => setThemeName(currentSystemTheme()), [setThemeName])

  useEffect(() => {
    systemThemeQuery.addEventListener('change', themeHandler)
    return () => systemThemeQuery.removeEventListener('change', themeHandler)
  }, [themeHandler])
```

Reply:
done